typed-racket-lib/typed-racket/typecheck/tc-structs.rkt

```diff
@@ -117,19 +118,30 @@
   (c:-> syntax? c:any/c (values identifier? (c:or/c Name? #f) (c:or/c Mu? Poly? Struct? Prefab? #f)))
   (syntax-parse nm/par
     [v:parent
-      (if (attribute v.par)
-          (let* ([parent0 (parse-type #'v.par)]
-                 [parent (let loop ((parent parent0))
-                               (cond
-                                 ((Name? parent) (loop (resolve-name parent)))
-                                 ((or (Poly? parent) (Mu? parent)
-                                      (if prefab? (Prefab? parent) (Struct? parent)))
-                                  parent)
-                                 (else
-                                  (tc-error/stx #'v.par "parent type not a valid structure name: ~a"
-                                                (syntax->datum #'v.par)))))])
-                (values #'v.name parent0 parent))
-          (values #'v.name #f #f))]))
+     (if (attribute v.par)
+         (let* ([parent-si-ins (syntax-local-value #'v.par (lambda () #f))]
+                [parent0 (parse-type
+                          (cond
+                            ;; maybe-struct-info-wrapper-type only returns parent's
+                            ;; type name when parent-si-ins is an TR's
```

Comment:
I'm not sure exactly what this comment means. Is it that this case happens when the structure is defined in a `#lang typed/racket` module?

Reply:
I think this also happens for `require/typed`.